examples/nlp/question_answering/question_answering_squad_infer.py

```diff
@@ -0,0 +1,300 @@
+        tokenizer.add_special_tokens(special_tokens)
+    else:
+        tokenizer_cls = nemo_nlp.data.NemoBertTokenizer
+        tokenizer_special_tokens = nemo_nlp.utils.MODEL_SPECIAL_TOKENS[args.model_type]
+        tokenizer_name = nemo_nlp.utils.MODEL_NAMES[args.model_type]["tokenizer_name"]
+        tokenizer = tokenizer_cls(
+            do_lower_case=args.do_lower_case,
+            pretrained_model=tokenizer_name,
+            special_tokens=tokenizer_special_tokens,
+            bert_derivate=args.model_type,
+        )
+
+    model_cls = MODEL_CLASSES[args.model_type]
+    model_name = nemo_nlp.utils.MODEL_NAMES[args.model_type]["model_name"]
+
+    if args.pretrained_model_name is None:
+        args.pretrained_model_name = model_name
+
+    if args.bert_config is not None:
+        with open(args.bert_config) as json_file:
+            config = json.load(json_file)
+        model = model_cls(**config)
+    else:
+        """ Use this if you're using a standard BERT model.
+        To see the list of pretrained models, call:
+        nemo_nlp.nm.trainables.huggingface.BERT.list_pretrained_models()
+        """
+        model = model_cls(pretrained_model_name=args.pretrained_model_name)
+
+    hidden_size = model.hidden_size
+
+    qa_head = nemo_nlp.nm.trainables.TokenClassifier(
+        hidden_size=hidden_size, num_classes=2, num_layers=1, log_softmax=False
+    )
+    if args.bert_checkpoint is not None:
+        model.restore_from(args.bert_checkpoint)
+
+    _, eval_output, eval_data_layer = create_pipeline(
+        data_file=args.data_file,
+        model=model,
+        head=qa_head,
+        max_query_length=args.max_query_length,
+        max_seq_length=args.max_seq_length,
+        doc_stride=args.doc_stride,
+        batch_size=args.batch_size,
+        version_2_with_negative=args.version_2_with_negative,
+        num_gpus=args.num_gpus,
+        batches_per_step=args.batches_per_step,
+        use_data_cache=not args.no_data_cache,
+    )
+    load_from_folder = None
+    if args.checkpoint_dir is not None:
+        load_from_folder = args.checkpoint_dir
+    evaluated_tensors = nf.infer(tensors=eval_output, checkpoint_dir=load_from_folder, cache=True)
+    unique_ids = []
+    logits = []
+    for t in evaluated_tensors[0]:
+        unique_ids.extend(t.tolist())
+    for t in evaluated_tensors[1]:
+        logits.extend(t.tolist())
+
+    start_logits, end_logits = np.split(np.asarray(logits), 2, axis=-1)
+
+    (all_predictions, all_nbest_json, scores_diff_json) = eval_data_layer.dataset.get_predictions(
+        unique_ids=unique_ids,
+        start_logits=start_logits,
+        end_logits=end_logits,
+        n_best_size=args.n_best_size,
+        max_answer_length=args.max_answer_length,
+        version_2_with_negative=args.version_2_with_negative,
+        null_score_diff_threshold=args.null_score_diff_threshold,
+        do_lower_case=args.do_lower_case,
+    )
+
+    if args.output_prediction_file is not None:
```

Comment:
can we make with argument required? there is no else

Reply:
If the user is not interested in the prediction e.g. for training, its better not the save a file to the users work space.